### ISSUE TITLE
Is it possible to get the distance between two samples?

### ISSUE THREAD

**Issue by paulaceccon** [tensorflow/similarity#189]
Suppose I train the model like [this](https://github.com/tensorflow/similarity/blob/master/examples/supervised_visualization.ipynb).
I'm trying to figure out if I could pass two new samples to the model and not retrieve the closest items, but to the distance between them so I can say whether they are the same or not (for that I'd also need to know the mean distance between the clusters and withing samples of the cluster, to find a good threshold), as can be seen [here](https://keras.io/examples/vision/siamese_contrastive/), under **Visualize the predictions**.

**Comment by owenvallis** [comment]
You can certainly do this using the following steps:

1.  Call model.predict() on a list of examples.
2.  Call tf.stack() on the list of predictions.
3.  Pass the stacked predictions to the same tensorflow_similarity.distances function used during training.
4.  This will produce a symmetric, pairwise distance matrix.

Alternatively, you can directly train a siamese network that computes the contrastive loss on the distance between two examples. This is not too hard, but will require writing a custom gradient.tape function. You can see something similar with the self-supervised model in the development branch.
